modules/cachedb_redis/cachedb_redis_dbase.c

```diff
@@ -382,7 +383,17 @@ redis_con* redis_new_connection(struct cachedb_id* id)
 	else
 		multi_hosts = 0;
 
-	r = parse_csv_record(_str(id->host));
+	LM_DBG("Redis host list: '%.*s' possible port '%d'\n", _str(id->host)->len, _str(id->host)->s, id->port);
+	if (multi_hosts == 0 && id->port != 0) {
+		host_w_port = pkg_malloc(sizeof(id->host) + sizeof((char *)":") + sizeof(id->port) + 1);
+		sprintf(host_w_port, "%s:%d", id->host, id->port);
+	}
+	else {
+		host_w_port = pkg_malloc(sizeof(id->host) + 1);
```

Comment:
* 2 x `pkg_malloc()` newly introduced
* 0 x `pkg_free()` newly introduced

So, at a first glance, this PR would introduce a memory leak while fixing the original issue.  So I will close it for now, as we can fix the port problem without allocating memory at all -- I will shortly push the commit on `master`.

Thank you for the effort!